test/integration/Test/Integration/Scenario/Wallets.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TypeApplications #-}
+
+
+module Test.Integration.Scenario.Wallets
+    ( spec
+    ) where
+
+import Prelude
+
+import Data.Aeson
+    ( Value )
+import Network.HTTP.Types.Status
+    ( status200 )
+import Test.Hspec
+    ( SpecWith, it )
+import Test.Integration.Framework.DSL
+    ( Context (..)
+    , Headers (..)
+    , Payload (..)
+    , expectResponseCode
+    , json
+    , request
+    )
+
+spec :: SpecWith Context
+spec = do
+    it "Create a wallet" $ \ctx -> do
+
+        let payload = Json [json| {
+                "name": "1st Wallet",
+                "mnemonic_sentence": ["identify", "screen", "lock", "bargain", "inch", "drop", "canyon", "flock", "dry", "zone", "wash", "argue", "system", "glory", "light"],
+                "mnemonic_second_factor": ["attract", "tornado", "slender", "pumpkin", "clown", "announce", "term", "winner", "ready"],
+                "passphrase": "Secure Passphrase",
+                "address_pool_gap": 20
+                } |]
+        response <- request @Value ctx ("POST", "v2/wallets") Default payload
+        print (show response)
+        expectResponseCode @IO status200 response
```

Comment:
Currently the response after creating a wallet is:
```
{
  "passphrase": {
    "last_updated_at": "2019-04-10T06:57:44.174387543Z"
  },
  "address_pool_gap": 20,
  "state": {
    "status": "restoring",
    "progress": {
      "quantity": 0,
      "unit": "percent"
    }
  },
  "balance": {
    "total": {
      "quantity": 0,
      "unit": "lovelace"
    },
    "available": {
      "quantity": 0,
      "unit": "lovelace"
    }
  },
  "name": "Piotr Wallet",
  "delegation": {
    "status": "not_delegating"
  },
  "id": "4bba596017a63e52a4cf6caaadf8a670bfe4745b"
}
```
which is more or less inline with the spec (http://localhost:8000/specifications/api/#operation/postWallet) - (the order of the items in JSON response is different in the spec)

The response code is `200` but `202` is in spec.

Reply:
The success response code is a mistake from our `Servant` definition. I wonder if we couldn't catch that automatically with unit tests :thinking: